### DISCUSSION TITLE
License implications of AIR SDK in the AUR (Arch Linux User Repository)

### DISCUSSION THREAD

**Discussion by B-Interactive** [airsdk/Adobe-Runtime-Support#2715] [General]
I've prepared an [AUR](https://aur.archlinux.org/) entry for the Harman AIR SDK, but before pushing, I want to confirm that I'm not stepping on toes.

The installer will:
- Install JDK 11 dependency.
- Download the AIR SDK from official source and install it to `/opt/air-sdk/[version]/`.
- Install the AIR SDK license under `/usr/share/licenses/air-sdk/LICENSE.pdf`.
- Set system wide AIR_HOME and JAVA_HOME environment variables, and add AIR SDK `bin` to PATH.

**The crux of my question:** The user wouldn't be required to accept the license agreement before the SDK will download, which is different behaviour to [here](https://airsdk.harman.com/download) and [here](https://airsdk.dev/docs/basics/install/linux).  The license is referenced and installed.  Is this going to be an issue?

Having this in the AUR, I'm hoping, will then permit me to approach Moonshine-IDE, possibly as an AUR entry too, as the AIR SDK is needed to run it under Linux.

**Comment by marchbold** [comment]
In our experience Harman requires acceptance of the license before downloading so yeah I'd imagine this will be an issue. Was definitely the reason we implemented those additional steps on the airsdk.dev site.

**Comment by B-Interactive** [comment]
Thanks @marchbold, that's been helpful.  I'll see what options there might be in the ways of prompting the user to accept, before continuing install.  

I might see if I can find an AUR precedent for that sort of prompt.

**Reply by marchbold** [reply]
Definitely a good idea, also worth reaching out to @ajwfrost to ensure they are okay with this as well.

**Comment by B-Interactive** [comment]
Is here the best place to check with @ajwfrost ?

I now have it presenting the following prompt, _requiring_ acceptance before it will download the SDK:

```
Harman AIR SDK License Agreement
https://airsdk.harman.com/assets/pdfs/HARMAN%20AIR%20SDK%20License%20Agreement.pdf

I accept the terms of the AIR SDK License Agreement [y/n]
```

If I enter `y`, download and installation continues.
If I enter `n`, download and installation is aborted.

**Reply by marchbold** [reply]
He normally responds here.

**Reply by ajwfrost** [reply]
Yes but UK timezone :-)

Thanks both: yes, our legal folk advised that we need to have people accept the terms of the license agreement before they actually download the software, since our agreement with Adobe requires us to only distribute to people who have accepted the license terms. They also said it needed to be a proactive step rather than just wording saying that "by downloading, you accept this".

Other than that -> thank you, that sounds great!

Moonshine I thought already provided the ability to download the AIR SDK..? Not tried it under Linux myself though..

thanks

**Reply by B-Interactive** [reply]
Thank you @ajwfrost .  UK time was fine, all these things were cleared up very quickly, thank you both!

Moonshine has an SDK installer, however to run Moonshine under Linux, one needs the AIR SDK (there is no runtime).  Cart, horse, etc :sweat_smile:

**Reply by ajwfrost** [reply]
Slightly curious how you're using the AIR SDK to run Moonshine (unless you're just copying an installation across from a Win/Mac machine and then using ADL?) ..  but I hadn't actually realised that it wasn't available already on Linux.

Anyway -> if you need any info/support, you can reach them on https://github.com/Moonshine-IDE/Moonshine-IDE/discussions

**Reply by B-Interactive** [reply]
I'm not yet running Moonshine IDE @ajwfrost, I would like to try though (and under Linux specifically).  In assembling the pieces I'd need to attempt it, I noticed the AIR SDK wan't in the Arch User Repository, so I took that on as a project as well.

Currently I'm not sure how AIR SDK 50 will play with Flex SDK 4.16.1 under Linux.  That's my current hurdle.

**Comment by B-Interactive** [comment]
For those interested in running on Arch Linux, the AIR SDK AUR entry is here:
https://aur.archlinux.org/packages/air-sdk

**Comment by B-Interactive** [comment]
Hello again @ajwfrost.  I'm receiving what might become some pressure from the Arch Linux side, to remove any prompts given to the user to accept the terms of the license agreement.  The worst case scenario is threatened complete removal of AIR SDK package from the Arch User Repository (AUR).

The reason offered is detailed in the Arch Wiki [note](https://wiki.archlinux.org/title/Creating_packages#PKGBUILD_variables) here:
> **Note:** *makepkg*, and thus the `build()` and `package()` functions, are intended to be non-interactive. Interactive utilities or scripts called in those functions may break *makepkg*

I wanted to check in with you, in case the agreement with Adobe may not have since become a bit more... flexible, no longer requiring users to **interactively** and proactively agree to the license agreement, before download commences.  Might a clearly worded non-interactive message before and/or after installation suffice, for example?

**Reply by ajwfrost** [reply]
Hi

I suspect they'd still have the same thinking for this, although it might be we could suggest a text output followed by a pop-up message when running any of the SDK tools for the first time (a bit like how the AIR installer works).

Although before I check with the legal team here, can I ask whether an alternative approach might be to use the AUR installation system to install the "AIR SDK Manager" tool? Since when this is then run, it could be used to download and manage the AIR SDK installations, and we could have this display the license prior to the downloads (currently, on Windows and macOS, the AIR SDK Manager has an installer which will ensure someone accepts the license agreement for the SDK there - but on Linux we are just looking at this as a zipped-up bundle, so that should fit with the "non-interactive" model.

This would make it a two-step process:
1) use the AUR system to download the AIR SDK Manager, non-interactive
2) run the AIR SDK Manager to download the latest/required AIR SDK, needing a UI and license acceptance

Would that be an acceptable alternative?

thanks

**Reply by B-Interactive** [reply]
Thanks @ajwfrost.  Yes the AIR SDK Manager sounds like it would be a great solution.

**Reply by B-Interactive** [reply]
Is there a Linux version available at this time?

**Reply by marchbold** [reply]
Linux version is in final testing.

**Reply by B-Interactive** [reply]
To take the pressure off in the meantime, I've adjusted the Arch AUR package so it no longer downloads the source file on behalf of the user.  They are instead instructed where to go to download it, and where to place it so the package can build and install correctly.

**Comment by dlukt** [comment]
Neither air-sdk nor adobe-air-sdk are present in the aur anymore.
What happened?

**Reply by B-Interactive** [reply]
I requested the air-sdk package I was maintaining be removed from the AUR after the AIR SDK Manager replaced direct AIR SDK downloads.  That air-sdk package had caused a bit of a stir in Arch land too, as I had been prompting users to agree with terms before downloading (as required by Harman), which goes against Arch packaging guidelines.

I've not picked up the reigns with the AIR SDK Manager, as I do not currently have an AIR licence, which is required to test under Linux.  That was causing issues even when I was managing the air-sdk.  If someone else is happy to test, I could try assembling an AUR package for the AIR SDK Manager.

**Reply by dlukt** [reply]
Thanks for explaining it.

I just needed it for Devolo's Cockpit app.
It's having issues with being 32bit and running of 64bit air.
I thought perhaps the PKGBUILD is available somewhere.
